echidna/limit/test_statistic.py

```diff
@@ -0,0 +1,269 @@
+""" Module to hold classes for various test statistics that can be used
+for fitting.
+
+"""
+import numpy
+
+import abc
+
+
+class TestStatistic(object):
+    """ Base class for the calculation of a test statistic.
+
+    The calculation of any test statistic is based on one spectrum
+    containing observed events and one containing expected events. It
+    is assumed that the observed events form the "data" spectrum and
+    the expected events form the spectrum predicted by the model.
+
+    Args:
+      type_name (string): Type of test statistic.
+      spectra_par (:class:`echidna.core.spectra.SpectraParameter): The
+        parameter class for the projection dimension e.g.
+        ``energy_mc``.
+
+    Attributes:
+      _type (string): Type of test statistic.
+      _spectra_par (:class:`echidna.core.spectra.SpectraParameter): The
+        parameter class for the projection dimension e.g.
+        ``energy_mc``.
+      _per_bin (:class:`numpy.ndarray`): 1D numpy array to hold value
+        of test statistic calculated for each bin
+
+    """
+    __metaclass__ = abc.ABCMeta  # Only required for python 2
+
+    def __init__(self, type_name, spectra_par):
+        self._type = type_name
+        self._spectra_par = spectra_par
+        self._per_bin = numpy.zeros(shape=(spectra_par._bins))
+
+    def get_type(self):
+        """
+        Returns:
+          string: Type of test statistic, stored in :attr:`_type`
+        """
+        return self._type
+
+    def get_spectra_par(self):
+        """
+        Returns:
+          (:class:`echidna.core.spectra.SpectraParameter):The parameter
+           class for the projection dimension, stored in
+           :attr:`_spectra_par`.
+        """
+        return self._spectra_par
```

Comment:
Dont think this is required as I'm already passing compute_statistic a projection of the ROI. This is not a spectra object just a ndarray